My final project

matiz86 opened the pull request.

matiz86 committed:
My final project

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+  </head>
+  <body class="page">
+    <header class="content header">
+      <a href="#" class="header__title">&lt;/html&gt;</a>
```
---

**Можно лучше:**

Логотип должен быть ссылкой без атрибута href(что бы по ссылке не было перехода)  для улучшения семантики страницы.

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+
+*/
+
+@supports (-webkit-text-stroke: 1px #fff) or (text-stroke: 1px #fff) {
```
**Нужно исправить:** Селектор с правилом @supports должен быть ниже чем селектор без правила, или тень через text-shadow всегда будет включена из-за того, что по специфике CSS, значение св-ва ниже перебивает значение св-ва выше.

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  color: var(--accent-color);
+  opacity: 0.5;
+  mix-blend-mode: hard-light;
+  -webkit-text-stroke: 1px var(--stroke-color);
```
Это св-во мы задаём через директиву @supports, тут оно лишнее

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  opacity: 0.5;
+  mix-blend-mode: hard-light;
+  -webkit-text-stroke: 1px var(--stroke-color);
+  text-shadow: 0 var(--stroke-color), 0 1px var(--stroke-color),
```
https://disk.yandex.ru/i/3VkG5uNoAKthEw При таком значении тень не будет отображаться, нужно поправить значение

Примеры и варианты значений этого свойства можно посмотреть поискав по названию свойства на

-  https://doka.guide/
-  https://developer.mozilla.org/ru/

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+}
+
+
+.card__image-label {
```
- Начертание заглавными буквами  должно быть задано через CSS св-во `text-transform: uppercase;`.

Avaakum commented on fonts/fonts.css:
```diff
@@ -1 +1,16 @@
-/* В этом файле вы импортируете все шрифты для сайта. Один из них вариативный и требует особенной настройки */
+@font-face {
+    font-family: 'PressStart2P-Regular';
```
Нужно переименовать шрифт без указания насыщенности. Всегда могут добавится ещё начертания.

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  width: 100%;
+}
+
+.card__like-button::before {
```
Используйте общий дополнительный класс для кнопок с одинаковыми стилями вместо дублирования стилей для разных кнопок

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  width: 100%;
+}
+
+.button-remember::before {
```
Используйте общий дополнительный класс для кнопок с одинаковыми стилями вместо дублирования стилей для разных кнопок

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  align-items: center;
+}
+
+.button-save::before {
```
Используйте общий дополнительный класс для кнопок с одинаковыми стилями вместо дублирования стилей для разных кнопок

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  font-variation-settings: 'wght' var(--text-weight);
+}
+
+.card__content-text:nth-child(1) {
```
Не следует использовать стилизацию по номеру блока, так как порядок блоков может измениться. Рекомендуется применять более гибкие подходы к стилизации, которые не зависят от конкретной нумерации блоков на странице, чтобы обеспечить устойчивость и адаптивность структуры. Более гибким решением будет воспользоваться вспомогательными классами для назначения “отличающийся” стилизации блока. А для однотипной стилизации использовать общую сущность.

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  margin-block-end: 27px;
+}
+
+.card__content-text:nth-child(2) {
```
Не следует использовать стилизацию по номеру блока, так как порядок блоков может измениться. Рекомендуется применять более гибкие подходы к стилизации, которые не зависят от конкретной нумерации блоков на странице, чтобы обеспечить устойчивость и адаптивность структуры. Более гибким решением будет воспользоваться вспомогательными классами для назначения “отличающийся” стилизации блока. А для однотипной стилизации использовать общую сущность.

Avaakum commented on styles/style.css:
```diff
@@ -1,20 +1,456 @@
+  min-block-size: 100dvb;
+
+  font-family: var(--main-font), sans-serif;
+  font-variation-settings: 'wght' 400;
```
- Насыщенность вариативного шрифта нужно задать через св-во font-variation-settings, ось 'wght'
 и  переменную
- Эта переменная должна иметь значение 465 для тёмных тем, для светлых 400

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+    <link rel="icon" href="./images/favicon.ico" sizes="any" />
+    <link class="fill-content" rel="icon" href="./svg/favicon.svg" type="image/svg+xml" />
+    <link rel="apple-touch-icon" href="./images/mobile-favicon.png" />
+    <meta name="color-scheme" content="dark light" />
```
---

**Отлично:**



 Цветовые схемы корректно описаны в теге

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+    </header>
+    <main class="content main">
+      <!-- Первая карточка -->
+      <article class="card">
```
---

**Отлично:**

Отличная семантика

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+      <!-- Первая карточка -->
+      <article class="card">
+        <h2 class="card__title">Фритрек и нулевой спринт: Подготовка к работе</h2>
+        <figure class="card__image">
```
Это классная семантика, но тут эта семантика не подходит, так как картинка является частью карточки, и не может существовать автономно.  Используйте div, а для подписи span

Тег `figure` представляет собой автономный блок контента. Это означает, что если вы передвинете блок в любое место документа, то это не повлияет на смысл документа . Поэтому его можно использовать не во всех ситуациях. 



 - [Правильный способ использовать элементы figure и figcaption](https://medium.com/@ABatickaya/%D0%BF%D1%80%D0%B0%D0%B2%D0%B8%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9-%D1%81%D0%BF%D0%BE%D1%81%D0%BE%D0%B1-%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D1%8C-%D1%8D%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82%D1%8B-figure-%D0%B8-figcaption-5e6e4786bdbd) хорошая статейка, чтобы разобраться, к сожалению только через vpn доступна в РФ
 - [Документация по figure](https://developer.mozilla.org/ru/docs/Web/HTML/Element/figure)

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+        <h2 class="card__title">Фритрек и нулевой спринт: Подготовка к работе</h2>
+        <figure class="card__image">
+          <img class="image" src="./images/card-image.png" alt="Машина" />
+          <figcaption class="card__image-label">&lt;/html&gt;</figcaption>
```
Для надписи используйте span

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+          <span class="button__text-rembemer">Сохранить на память</span>
+        </button>
+      </div>
+      <dialog class="dialog" id="dialog-id">
```
Лишний элемент для этого спринта

Avaakum commented on index.html:
```diff
@@ -1,46 +1,109 @@
+      </article>
+      <div class="button">
+        <button
+          type="button"
```
---

**Отлично:**

- любая кнопка которая не отправляет форму `<form>`  должна иметь **тип button**(атрибут type), потому что по умолчанию у всех копок тип `submit` для отправки формы, а если кнопка лежит вне формы и отправлять ничего никуда не должна.

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Здравствуйте! У вас получилась хорошая работа. _Пожалуйста, не торопитесь, и после исправления проверяйте исправление всех критических комментариев и соответствие макету на всех брейкпоинтах._


_Что понравилось:_

- цвет в символах кнопок заполняется корректно 
- Правильное форматирование
- Корректная структура HTML
- корректные анимации
- отличная структура файлов проекта
- Хорошая семантика
- Хорошие стили 
- Отзывчивая верстка
- Хорошо адаптирована вся страница

_Что необходимо исправить:_

- **Все комментарии  с пометкой "Нужно исправить" в коде**



- https://disk.yandex.ru/i/8f0dzQx2ZcHTuQ некорректная анимация сердца при добавлении в избранное. Сверьтесь с макетом и заданием








- https://disk.yandex.ru/i/YNu1WjdlT4gWCg  у этих кнопок должен быть одинаковый ховер и фокус. Для отдельного дополнительного класса нужно присвоить общие стили, в том числе и состояния при ховере и фокусе 



- **При исправлении для одной из тем (светлой/тёмной) обязательно проверяйте, чтобы при такой же автотеме (светлой/тёмной) было соответствующее исправление.** Это напоминалка, а не конкретное исправление.

- В проекте есть нарушения в отображении цветовых палитр в темной и светлой теме, цветовые палитры необходимо полностью проконтролировать
 - https://disk.yandex.ru/i/-gaCv_4wN0Gl2w // https://disk.yandex.ru/i/1GFLZMQCv3Bl7w фон окна хедера и карточки, и кнопки должен быть белый при светлой и авто-светой теме
 

_Что можно улучшить:_

- Все комментарии с пометкой "Можно лучше"
- CSS-правила в коде желательно распологать в примерном соответствии с позицией элементов в DOM.

- Для всех функций ` var() ` в CSS нужно задать резервные значения,  на случай если доступ к кастомному свойству по имени будет невозможен. 

- Про альтернативный текст для картинок https://habr.com/ru/post/341810/

- Можно использовать удобное сокращение  [background](https://developer.mozilla.org/en-US/docs/Web/CSS/background)
- Для единообразного форматирования документов в VSCode лучше положить файл `.editorconfig` в корневой каталог. Файл .editorconfig это файл для единого формата редактирования во всех языках, который поддерживает большинство IDE, что бы не нужно было в каждом новом проекте/языке  настраивать форматирование документов, а просто класть этот файлик в корень проекта. Ссылки по теме:
  - https://habr.com/ru/articles/220131/
  - https://editorconfig.org/
  - Пример файла https://disk.yandex.ru/d/Q7mSEMV3ua23Qg

- А так же можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`



**Напоминание:** После выполнения всех критических замечаний работа будет принята. Предложения по улучшению выполняются по желанию.

Желаю успеха!

matiz86 committed:
Second review

matiz86 committed:
Update v3

matiz86 committed:
Add files via upload

Замечания поправил. Единственное, я не очень понимаю какие проблемы с анимацией сердца ,и ховер-фокус  по кнопкам. Внимательно сравнил с UI kit фигма, так же рекомендации по верстке проекта, вроде все похоже и по анимации и по эффектам.

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Здравствуйте. К сожалению проект отклоняется от проверки. Итерация не теряется. 




- _На повторных итерациях не исправлены критические замечания._ 
    - **Пример:** 
              - Начиная с неисправленных пунктов  в коде: https://disk.yandex.ru/i/gBY9eZSLYJx5Mg // https://disk.yandex.ru/i/zVZQOX3CqsQbDA // https://disk.yandex.ru/i/ILx-5hRONl-EPg // https://disk.yandex.ru/i/Q64oCE3xjY3R7Q и д в ревью предыдущей итерации нужно проверить исправление **всех критических комментариев** в работе. Если у вас есть сомнения по поводу того, что комментарий исправлен, то пожалуйста, перечитайте комментарии внимательно, там описано, что нужно исправить в работе. Если что-то исправить не получается, то можно смело обращаться к наставнику, он сможет подсказать. Чтобы исправить ошибки валидатора(при наличии), можно в chrome перевести страницу на русский, чтобы было понятнее, что именно за ошибки он выдаёт. А для соответствия макету можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi  )  для Chrome.



     - Пожалуйста, будьте внимательны, проверяйте, исправляйте и присылайте работу только после исправления **всех критических замечаний** c предыдущей итерации
    - Если у вас есть технические проблемы, то попробуйте обратиться в поддержку или спросить в чате студентов не было ли у кого похожих случаев.




_Вне счета итераций(без исправления **критических ошибок** работу не сдать):_

- https://disk.yandex.ru/i/fUWXYONEcj3ydA Пропала кнопка "сохранить"

matiz86 committed:
final

matiz86 committed:
Add files via upload

matiz86 committed:
Add files via upload
